torchvision/models/detection/retinanet.py

```diff
@@ -578,14 +585,37 @@ def forward(self, images, targets=None):
         return self.eager_outputs(losses, detections)
 
 
-model_urls = {
-    "retinanet_resnet50_fpn_coco": "https://download.pytorch.org/models/retinanet_resnet50_fpn_coco-eeacb38b.pth",
-}
+class RetinaNet_ResNet50_FPN_Weights(WeightsEnum):
+    COCO_V1 = Weights(
+        url="https://download.pytorch.org/models/retinanet_resnet50_fpn_coco-eeacb38b.pth",
+        transforms=ObjectDetection,
+        meta={
+            "task": "image_object_detection",
+            "architecture": "RetinaNet",
+            "publication_year": 2017,
```

Comment:
You are right to say that `(1, 1)` is the right value but I felt that this is somehow misleading. It doesn't really matter what input size you provide to the model because the `GeneralizedRCNNTransform` will internally resize the image to 800 on the shortest edge and 1333 on the max. That's the reason I opted not to define it but I don't mind adding it; technically it will accept any size because internally it resizes. Thoughts?

Reply:
Then technically like SSD we should document this with size = (800, 1333) ?

I'm not sure. But somehow the aim is to not hide the fact what generalized RCNN does and what size the model expects / takes. Either the minimum size / like classification the size model takes.

Reply:
It's not going to be `(800, 1333)`. The `GeneralizedRCNNTransform` maintains the ratio of the image by setting the minimum edge to 800. Then if the maximum edge exceeds 1333 it resizes the image further. So the input size in the network is variable.

If we put `(1, 1)` as the `min_size`, that's technically accurate but it doesn't have the same interpretation as for the classification case. In classification this size depicts the actual size of the smallest image that can go through the network. Here the value `(1, 1)` doesn't tell us anything about the network; it just tells us that we can throw any size in because we will resize it eventually to at least 800.

More thoughts based on the above clarifications?

Reply:
I get it. I guess the best solution is what you are trying in #5444 . If we are able to separate out `GeneralizedRCNNTransform`. We can write clearer docs as what it does.

A small doubt though what does `size=(224,224)` in classification models then stands for? 
That's the size of image passed to them when the models were trained I think? Since even they are flexible due `AdaptiveAvgPool2d`?

Reply:
The `size` field gives you information about the inference size of the model. This is the size of the crop that the model expects. Whether this is fixed size or variable depends on the model; for instance ViT is not flexible.

Again I'm happy to make changes by adding, removing, amending the meta-data info to make them more readable/understandable. Just trying to provide clarifications of what each field means because currently we are lacking documentation (TBD). Let me know thoughts/ideas.

Reply:
Sounds good. We would need documentation on meta-data info. I will too think over it :)